plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -247,7 +247,29 @@ public synchronized Optional<Table> getTable(String databaseName, String tableNa
                 // TODO
                 break;
         }
-        throw new IllegalStateException("Unknown action type");
+        throw new IllegalStateException("Unknown action type: " + tableAction.getType());
+    }
+
+    public synchronized boolean isReadableWithinTransaction(String databaseName, String tableName)
+    {
+        Action<TableAndMore> tableAction = tableActions.get(new SchemaTableName(databaseName, tableName));
+        if (tableAction == null) {
+            return true;
+        }
+        switch (tableAction.getType()) {
+            case ADD:
+            case ALTER:
+                return true;
+            case INSERT_EXISTING:
+            case DELETE_ROWS:
+            case UPDATE:
+                // Until transaction is committed, the table data may or may not be visible.
+                return false;
+            case DROP:
+            case DROP_PRESERVE_DATA:
+                return false;
+        }
```

Comment:
Looks good to me from glancing at our HMS interactions however I'm not deeply familiar.